### ISSUE TITLE
Collection chooser does not store value

### ISSUE THREAD

**Issue by puredriver** [donhauser/wagtail-photo-gallery#2]
Hi Jonas,
the wagtail admin does not store the collection value. I can select a collection but when i press "save draft" of the page the collection value is empty  
Wagtail 2.15.1
Any ideas...thx
Andre

**Comment by donhauser** [comment]
Hi Andre,
thank you for your report :)

Its quite hard to tell what actually causes the fault but I'll try to give you some background information:

The `ImageGalleryMixin` enables the usage of `blocks.GalleryBlock` in wagtail pages.
The `GalleryBlock` hooks in instances of `Album` class into the page iff `GalleryBlock.collection` matches `Album.collection`.

The `Album` class is realized using **django-modelcluster** (it inherits from `ClusterableModel`).
From my experience, even though it is super useful, *django-modelcluster* can lead to pretty strange errors if the database is not as it should be. E.g. if you did not do all migrations etc.

So to come back to your problem:
There are not so many possible causes (as the collection attribute just occurs twice), as the collection does not occur that often.
Before you are digging into it further I would suggest that you check if the migrations were applied correctly

**Comment by puredriver** [comment]
Thx for your quick answer!
I tried it again with a fresh wagtail installation...described here: https://docs.wagtail.io/en/stable/getting_started/tutorial.html (until the admin is accessible) and your instructions in the README. Migrations were successfully done.


HomePage Model
```
class HomePage(ImageGalleryMixin, Page):
    content = StreamField([
        #...
        ("gallery", GalleryBlock()),
        #...
    ], blank=True)
    
    # content panel for the CMS (same as always)
    content_panels = Page.content_panels + [
        StreamFieldPanel("content"),
    ]
```

![Screenshot 2021-12-21 140422](https://user-images.githubusercontent.com/28884396/146935316-192834f4-19e3-4fcc-91e8-0f63a48b57ae.png)

Django 3.2.10
Wagtail 2.15.1

Unfortunatly same result see screenshot

**Comment by donhauser** [comment]
Hi again,
I was able track down the error and it is actually **related to wagtail>=2.13**.

The last version I've tested was wagtail 2.12, sorry for that.
I just started a fresh wagtail demo like you did to reproduce the mistake. It turned out that the collection attribute is stored correctly and is read out correctly, but the attribute is not available to the `AdminChooser.render_html()`.
After quite some research I found out that wagtail started to use some javascript rendering called [telepath](https://wagtail.github.io/telepath/).

The problem was that for wagtail > 2.12 every class inheriting from  `ChooserBlock` (from the extention [wagtail-generic-chooser](https://github.com/wagtail/wagtail-generic-chooser) needs to implement a `get_form_state()` function. The `GalleryBlock` is `StructBlock` that contains such an affected block `CollectionChooserBlock`.

To fix the problem `CollectionChooserBlock` (in *blocks.py*) needs to be extended by the following code:
```py
    def get_form_state(self, value):
        return self.widget.get_value_data(value)
```
I will update this package to fix this bug.
